scripts/mixins/job_special.lua

```diff
@@ -0,0 +1,52 @@
+-- for mobs with job-specific special ability
+
+require("scripts/globals/mixins");
+require("scripts/globals/status");
+
+g_mixins = g_mixins or {};
+
+g_mixins.job_special = function(mob)
+
+    mob:addListener("SPAWN", "JOB_SPECIAL_SPAWN", function(mob)
+        local defaultAbility = {
+            [JOBS.WAR] = jobSpec.MIGHTY_STRIKES,
+            [JOBS.MNK] = jobSpec.HUNDRED_FISTS,
+            [JOBS.WHM] = jobSpec.BENEDICTION,
+            [JOBS.BLM] = jobSpec.MANAFONT,
+            [JOBS.RDM] = jobSpec.CHAINSPELL,
+            [JOBS.THF] = jobSpec.PERFECT_DODGE,
+            [JOBS.PLD] = jobSpec.INVINCIBLE,
+            [JOBS.DRK] = jobSpec.BLOOD_WEAPON,
+            [JOBS.BST] = jobSpec.FAMILIAR,
+            [JOBS.BRD] = jobSpec.SOUL_VOICE,
+            [JOBS.SAM] = jobSpec.MEIKYO_SHISUI,
+            [JOBS.NIN] = jobSpec.MIJIN_GAKURE,
+            [JOBS.DRG] = jobSpec.CALL_WYVERN,
+            [JOBS.SMN] = jobSpec.ASTRAL_FLOW,
+        };
+
+        mob:setLocalVar("usedJobSpec", 0);
+        mob:setLocalVar("useJobSpecAtPct", math.random(40,60));
+        local ability = defaultAbility[mob:getMainJob()];
+        if (mob:getLocalVar("jobSpec") == 0 and ability ~= nil) then
+            mob:setLocalVar("jobSpec", ability);
+        end
+    end);
+    
+    mob:addListener("WEAPONSKILL_STATE_ENTER", "JOB_SPECIAL_WS", function(mob, skillID)
+        if (mob:getHPP() < mob:getLocalVar("useJobSpecAtPct") and mob:getLocalVar("usedJobSpec") == 0) then
+            local spec = mob:getLocalVar("jobSpec");
+            if (spec > 0) then
+                mob:useMobAbility(spec);
+                mob:setLocalVar("usedJobSpec", 1);
+            end
+        end
+    end);
+    
+    mob:addListener("DEATH", "JOB_SPECIAL_DEATH", function(mob)
+        mob:setLocalVar("usedJobSpec", 0);
```

Comment:
i'm... 85% sure that local vars get all reset on spawn, so this is probably unnecessary

Reply:
they do get wiped.

Reply:
huh. i knew that, and added that code anyway. don't know why.  removed death listener.